spec/workflow/states/pass_spec.rb

```diff
@@ -89,5 +89,48 @@
         )
       end
     end
+
+    context "Without Results" do
+      let(:input) { {"color" => "red"} }
+      let(:payload) do
+        {"Pass" => {"Type" => "Pass", "End" => true}}
+      end
+
+      it "passes output through to input" do
+        workflow.run_nonblock
+        expect(ctx.output).to eq(input)
+      end
+
+      context "with InputPath" do
+        let(:payload) do
+          {"Pass" => {"Type" => "Pass", "End" => true, "InputPath" => "$.color"}}
+        end
+
+        it "Uses InputPath to select color" do
+          workflow.run_nonblock
+          expect(ctx.output).to eq("red")
```

Comment:
So I think this is wrong?  I think this should be

```suggestion
          expect(ctx.output).to eq("\"red\"")
```

Reply:
Yeah I think you're right and that matches what the asl sim outputs from the description.

Reply:
I think this is a slightly different issue than the one I'm trying to fix here though.  The core issue for this PR is we were completely ignoring InputPath, the issue you describe is around JSON and workflow/state input/output.  Think that is probably better solved in https://github.com/ManageIQ/floe/pull/222

Reply:
I feel all values should be ruby objects
And only the transitions to the outside world should convert json

So `CLI` and `Task` (converting their output to our `Result`) do conversion

Reply:
Yeha I was conflicted on whether context.output was in Ruby still or should already be converted to a JSON string.

Reply:
@Fryguy I think https://github.com/ManageIQ/floe/pull/228/commits/b7377a89a6638bb1743cd35c174a12c2e57acbb0#diff-c4b20f6026d80247d4c5645a2811f66f0bd918ebb4763da5fb0d19c8a7d092fa covers your concern here, still very WIP but throw :tomato:

Reply:
Yeah I thought about it a bit more and I agree with @kbrock here, context is "internal" where workflow.output is defining the "user interface".  As long as the states get the right input/output and the workflow handles input/output properly everything is up to us.

I changed https://github.com/ManageIQ/floe/pull/228 to have workflow.output as JSON